include/deal.II/non_matching/coupling.h

```diff
@@ -0,0 +1,348 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_non_matching_coupling
+#define dealii_non_matching_coupling
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/quadrature.h>
+
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/grid_tools_cache.h>
+
+#include <deal.II/fe/component_mask.h>
+#include <deal.II/fe/mapping_q1.h>
+#include <deal.II/fe/fe.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/lac/constraint_matrix.h>
+
+
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+namespace NonMatching
+{
+  /**
+   * Create a coupling sparsity pattern for non-matching, overlapping grids.
+   *
+   * Given two non-matching triangulations, representing the domains $\Omega$
+   * and $B$, with $B \subseteq \Omega$, and two finite element spaces
+   * $V(\Omega)$ and $Q(B)$, compute the sparsity pattern that would be
+   * necessary to assemble the matrix
+   *
+   * \f[
+   * M_{ij} := \int_{B} v_i(x) w_j(x) dx, \quad \foreach v_i \in V(\Omega), w_j \in Q(B)
```

Comment:
mathematically speaking, I think it would be clearer to write "\foreach 0\le i\le \text{dim}\; V(\Omega), ...(same for j)" since you're not actually iterating over all v_i that are members of V, but specifically the *basis* functions of that space.